storage/onpremise/mongodb-sharded/versions.tf

```
@@ -0,0 +1,25 @@
+terraform {
+  required_version = ">= 1.2"
```

Optional types are available only since terraform 1.3

```suggestion
  required_version = ">= 1.3"
```